pkg/kubelet/kubelet.go

```diff
@@ -1556,11 +1561,43 @@ func (kl *Kubelet) podFieldSelectorRuntimeValue(fs *api.ObjectFieldSelector, pod
 	}
 	switch internalFieldPath {
 	case "status.podIP":
-		return pod.Status.PodIP, nil
+		return kl.getPodIP(pod)
 	}
 	return fieldpath.ExtractFieldPathAsString(pod, internalFieldPath)
 }
 
+// getPodIP lazily finds the pod IP by consulting the following sources in order:
+//
+// 1.  The status of the passed in pod on the pod itself
+// 2.  The status of the pod in the kubelet status manager
+// 3.  The result of 'inspect' from the container runtime
+//
+// getPodIP should only be used in situations where the pod IP may not be
+// known by the status manager, such as starting containers.  Ordinarily, the
+// status manager is the source of truth for pod IP.
+func (kl *Kubelet) getPodIP(pod *api.Pod) (string, error) {
+	if len(pod.Status.PodIP) != 0 {
+		return pod.Status.PodIP, nil
+	}
+
+	status, ok := kl.statusManager.GetPodStatus(pod.UID)
```

Comment:
cc/ @ncdc 

@yujuhong and I discussed this offline after our sig-node meeting. For each SyncPod iteration, Kubelet will check InfraContainer status first including podIP too. Passing PodIP to call stack is the right approach for now.